qt/python/mantidqt/widgets/codeeditor/interpreter.py

```diff
@@ -150,6 +154,80 @@ def set_editor_readonly(self, ro):
     def set_status_message(self, msg):
         self.status.showMessage(msg)
 
+    def replace_tabs_with_spaces(self):
+        if self.editor.selectedText() == '':
+            self.editor.selectAll()
+        new_text = self.editor.selectedText().replace('\t', '    ')
```

Comment:
Can we extract out the `\t` and whitespace strings here as class-level variables to avoid hard-coding strings in the middle of the code?
Also, would it be possible to extract the functionality here and in `replace_spaces_with_tabs` as they are practically identical. Maybe a method that takes the 2 parameters `(match_text, replace_text)` that can be called by both this and `replace_spaces_with_tabs`?